daemon/src/position_metrics.rs

```diff
@@ -158,31 +161,28 @@ impl Cfd {
         use EventKind::*;
         match event.event {
             ContractSetupStarted => Self {
-                is_open: false,
-                is_closed: false,
-                is_failed: false,
-                is_refunded: false,
+                state: AggregatedState::New,
                 ..self
             },
             ContractSetupCompleted { .. } | LockConfirmed => Self {
-                is_open: true,
+                state: AggregatedState::Open,
                 ..self
             },
             ContractSetupFailed => {
                 // This is needed due to a bug that has since been fixed; `OfferRejected` and
                 // `ContractSetupFailed` are mutually exclusive.
                 // We give `OfferRejected` priority over `ContractSetupFailed`.
-                if self.is_rejected {
-                    Self { ..self }
+                if let AggregatedState::Rejected = self.state {
```

Comment:
I would just set it to `Failed`. I don't think we gain anything by reasoning about the previous state.

Reply:
Isn't this the bug what Daniel described?

Reply:
Fair enough: wrongly reporting failed can be problematic because we might think there's a problem when looking at the metrics.

On the other hand, this only applies for CFDs in the `cfds` table. As soon as we move CFDs to the `failed_cfds` table, this problem goes away. That combined with the fact that the comment says that the bug has already been fixed makes me think that we don't need any of this code (or the comment).

Reply:
I'll remove it in a separate commit.